client/app/stores/message_store.coffee

```diff
@@ -360,18 +361,31 @@ class MessageStore extends Store
         handle ActionTypes.MESSAGE_UNDO_TIMEOUT, ({ref}) ->
             delete _undoable[ref]
 
-        handle ActionTypes.MESSAGE_FETCH_REQUEST, ({mailboxID}) ->
+        handle ActionTypes.MESSAGE_FETCH_REQUEST, ->
+            return if not (url = @getNextUrl()) or @isFetching()
+
             # There may be more than one concurrent fetching request
             # so we use a counter instead of a boolean
-            _fetching++
-            @emit 'change'
+            mailboxID = AccountStore.getSelectedMailbox()?.get 'id'
+            ts = Date.now()
+            XHRUtils.fetchMessagesByFolder url, (err, rawMsg) ->
```

Comment:
@aenario : `MessageActionCreator` should be more appropriate than `accountActionCreator`?